### DISCUSSION TITLE
`scheduler.tools.run_job` stealing visibility of real callables

### DISCUSSION THREAD

**Discussion by gabriels1234** [dsoftwareinc/django-rq-scheduler#108] [General]
Hi @cunla ,

Besides from the current dependency on django-rq which makes truly cumbersome to troubleshoot failed jobs (they all look the same), the visibility is also very affected on every rq-related tool (e.g logging platform) that thinks that the callable is the actual callable, and not a helper to call the actual callable.
I understand that the need for scheduler.tools.run_job comes from being able to use functions as parameters that need to be evaluated before running the job each time.

So, I propose (and I offer my help to do), make it so if there's no callable function as argument, then the `True` callable is put. and if there's a callable then the scheduler will schedule scheduler.tools.run_job.

Let me know.

**Comment by gabriels1234** [comment]
Upon further look, it feels like a deal-breaker to have all jobs (especially the failing ones) named `scheduler.tools.run_job`.

Here I have, another easier-to-implement idea:
make the job id something related to the Job Name. namely `job_id=f"{job-name-separated-by-dash}-{random string}"`.

Let me know if you need help with it.

One dirty solution is to make the Cronjob's job be a job that enqueues the real job (at front). and the Cronjob result_ttl would be 1 or even 0.

Thanks!

**Comment by cunla** [comment]
This led me to work on removing the dependency on django-rq and rq (among other things, but like you said, that was the deal breaker).

I am working on redesigning it at the moment, and the more I dig into it, I realize it is better to write a new package from scratch.
If you want, we can discuss you maintaining django-rq-scheduler while I develop the new package.

**Reply by gabriels1234** [reply]
I like that!

**Reply by gabriels1234** [reply]
I made some changes and for now, I created a "run" function that receives the job/function to run . it set to have a result_ttl of 0 so in the list of jobs, the callable will look familiar/identifiable.
I also made it so the job_id is a part of the callable and a part of uuid4(8 chars).

I also added some helper functions to: requeue all jobs in a registry, etc. (but not sure if you'd like it in django-rq-scheduler) (that are callable so you can have jobs that fix common issues the end-users of the scheduler get into)
Let me know!

**Reply by cunla** [reply]
Please book 30mins to discuss how we proceed with this: https://cal.com/d/bSyDRNqssBV4Zd8AxU1sJt/oss
